robot/web/stream/server.go

```diff
@@ -82,90 +47,85 @@ func (ss *streamState) Stop() {
 	}
 }
 
-type peerState struct {
-	stream  *streamState
-	senders []*webrtc.RTPSender
-}
+// Server implements the gRPC audio/video streaming service.
+type Server struct {
+	streampb.UnimplementedStreamServiceServer
 
-type streamServer struct {
 	mu                      sync.RWMutex
 	streams                 []*streamState
-	nameToStream            map[string]Stream
+	nameToStream            map[string]gostream.Stream
 	activePeerStreams       map[*webrtc.PeerConnection]map[string]*peerState
 	activeBackgroundWorkers sync.WaitGroup
 }
 
-func (ss *streamServer) ServiceServer() streampb.StreamServiceServer {
-	return &streamRPCServer{ss: ss}
+// NewServer returns a server that will run on the given port and initially starts with the given
+// stream.
+func NewServer(streams ...gostream.Stream) (*Server, error) {
+	ss := &Server{
+		nameToStream:      map[string]gostream.Stream{},
+		activePeerStreams: map[*webrtc.PeerConnection]map[string]*peerState{},
+	}
+	ss.mu.Lock()
+	defer ss.mu.Unlock()
```

Comment:
This is mostly for my edification, I don't think how this is will hurt anything:
Why do we need to lock `mu` here? 

No other thread has access to `ss` outside of this function (as we just created it). Is there really any risk of a data race if we write to streams w/o a mutex before returning the `ss` pointer?